trade-finance-manager-api/src/v1/controllers/amendment.controller.js

```diff
@@ -89,14 +89,15 @@ const createAmendmentTFMObject = async (amendmentId, facilityId, auditDetails) =
     // gets latest amendment value and dates
     const latestValueResponse = await api.getLatestCompletedAmendmentValue(facilityId);
     const latestCoverEndDateResponse = await api.getLatestCompletedAmendmentDate(facilityId);
+    const facility = await api.findOneFacility(facilityId);
 
     let tfmToAdd = {};
     // populates array with latest value/exposure and date/tenor values
     tfmToAdd = await addLatestAmendmentValue(tfmToAdd, latestValueResponse, facilityId);
     tfmToAdd = await addLatestAmendmentCoverEndDate(tfmToAdd, latestCoverEndDateResponse, facilityId);
 
     let latestFacilityEndDateResponse;
-    if (isTfmFacilityEndDateFeatureFlagEnabled()) {
+    if (isTfmFacilityEndDateFeatureFlagEnabled() && isGefFacilityType(facility.facilitySnapshot.type)) {
```

Comment:
can these functions be declared as a const before? eg `const facilityEndDateFlag = isTfmFacilityEndDateFeatureFlagEnabled()` and const `isGefFacility = isGefFacilityType(facility.facilitySnapshot.type)` and those constants be placed in the if statement -> if (facilityEndDateFlag && isGefFacility ) {

Reply:
I've moved out `isGefFacility = isGefFacilityType(facility.facilitySnapshot.type)`, but I think it's fine to keep `isTfmFacilityEndDateFeatureFlagEnabled()` as an in-line check as the naming is clearer than pulling it out to a shorter name